Adding redux, Removing zoom block and moving to ES6

avivbueno opened the pull request.

avivbueno committed:
Adding redux and Removing zoom block

avivbueno committed:
Fix slow filter

eladg reviewed (commented):
looks good, not a big progress though. Writing you on messenger

eladg reviewed (changes requested):
Redux should be handled on a different file

avivbueno committed:
Moving Redux to a diffrent file + adding a meta tag for mobile support

eladg commented on src/js/reducers.js:
```diff
@@ -0,0 +1,14 @@
+      if (state.apiData === undefined) {
+        return state;
+      }
+      state.apiData = action.apiData;
```
This should be in a function

eladg replied:
state is immutable, you should not update it in this way

avivbueno replied:
@eladg Moved to a new file, take a look. (data.js)

eladg commented on src/js/reducers.js:
```diff
@@ -0,0 +1,14 @@
+        return state;
+      }
+      state.apiData = action.apiData;
+      return state;
```
you should not return `state` but the actual state, something like:
```javascript
{ apiDate: payload.data } 
```

avivbueno replied:
@eladg fixed this in the new method in the new file (data.js)

avivbueno committed:
Adding webpack

avivbueno committed:
Moving to Babel and Webpack

avivbueno committed:
Connecting webpack with CI

avivbueno committed:
Auditing npm install time

avivbueno committed:
Doing Semantic Versioning Correct

MAJOR.MINOR.PATCH

MAJOR - version when you make incompatible API changes,
MINOR - version when you add functionality in a backwards-compatible manner, and
PATCH - version when you make backwards-compatible bug fixes.
